networkx/generators/tests/test_stochastic.py

```diff
@@ -51,6 +51,16 @@ def test_multidigraph(self):
             (0, 2, d),
         ]
 
+    def test_zero_weights(self):
+        G = nx.DiGraph()
+        G.add_edge(0, 1, weight=0)
+        G.add_edge(0, 2, weight=0)
+        S = nx.stochastic_graph(G)
+        assert sorted(S.edges(data=True)) == [
+            (0, 1, {"weight": 0}),
+            (0, 2, {"weight": 0}),
```

Comment:
Thanks for the pull request @Emmanuel-Lud! but we are supposed to test the degree == 0.
And it's also possible that we don't even need to do it as this bit of code block isn't ever reached
as caught in https://github.com/networkx/networkx/pull/6069

I'll go ahead and close this PR. Feel free to fix another open issue :)